pkg/kv/kvserver/rangefeed/buffered_sender.go

```diff
@@ -111,6 +95,25 @@ type BufferedSender struct {
 	metrics *BufferedSenderMetrics
 }
 
+type streamState int64
+
+const (
+	// streamActive is the default state of the stream.
+	streamActive streamState = iota
+	// streamOverflowing is the state we are in when the stream has reached its
+	// limit and is waiting to deliver an error.
+	streamOverflowing streamState = 1 << (iota - 1)
```

Comment:
why the fancy shift?

Reply:
Heh, I was originally thinking there would be a need to know multiple things at once about the stream but then decided I didn't need that. I'll change this.